tests/test_m2m_100.py

```diff
@@ -0,0 +1,70 @@
+import unittest
+
+from tests.methods.test_config_union import ConfigUnionAdapterTest
+from transformers import M2M100Config
+from transformers.testing_utils import require_torch
+
+from .composition.test_parallel import ParallelAdapterInferenceTestMixin, ParallelTrainingMixin
+from .methods import (
+    BottleneckAdapterTestMixin,
+    CompacterTestMixin,
+    IA3TestMixin,
+    LoRATestMixin,
+    PrefixTuningTestMixin,
+    ReftTestMixin,
+    UniPELTTestMixin,
+)
+from .test_adapter import AdapterTestBase, make_config
+from .test_adapter_backward_compability import CompabilityTestMixin
+from .test_adapter_conversion import ModelClassConversionTestMixin
+from .test_adapter_embeddings import EmbeddingTestMixin
+from .test_adapter_fusion_common import AdapterFusionModelTestMixin
+from .test_adapter_heads import PredictionHeadModelTestMixin
+
+
+class M2M100AdapterTestBase(AdapterTestBase):
```

Comment:
Since this model doesn't support text classification tasks (and we test text classification training by default in the test runs), we'd have to override the `add_heead()` and `dataset()` methods here. That would roughly look like here: https://github.com/adapter-hub/adapters/blob/f0ca96200d40138e3e2b541ce9789e1119131a99/tests/test_bert_generation.py#L29-L70